Source/WebCore/platform/graphics/GraphicsLayer.cpp

```diff
@@ -575,7 +576,8 @@ void GraphicsLayer::paintGraphicsLayerContents(GraphicsContext& context, const F
         clipRect.move(offset);
     }
 
-    client().paintContents(this, context, clipRect, layerPaintBehavior);
+    if (!m_headlessModeEnabled)
```

Comment:
This just stops painting. You probably want to also avoid backing store allocation. I think the best way to do that is to make every layer behave as if `m_drawsContent` is false if the clients says to.